### PR TITLE
[Modular][Ready] Cyborg Tipping modularization

### PR DESCRIPTION
**Pull request by desertmote**
## About The Pull Request

Modularizes cyborg tipping.

## Changelog
JohnFulpWillard and theOOZ

### CODE REVIEW

**Comment by Gandalf2k15** [comment]
Aaah non-modular changes, pls fix

**Comment by desertmote** [comment]
> Aaah non-modular changes, pls fix

it is because this PR has [10740](https://github.com/Skyrat-SS13/Skyrat-tg/pull/10740) inside of it.

**Comment by theOOZ** [comment]
It's not gonna pass CI until #10740 is merged, which is not passing CI until the reverts here are merged, funny.

File: modular_skyrat/modules/altborgs/code/robot_tipping.dm [hunk 1]
```diff
@@ -0,0 +1,73 @@
+//	Modular solution for alternative skins and tipping
+/mob/living/silicon/robot/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/tippable/skyrat, \
+		tip_time = 3 SECONDS, \
+		untip_time = 2 SECONDS, \
+		self_right_time = 60 SECONDS, \
```
Comment (KathrinBailey) [inline modular_skyrat/modules/altborgs/code/robot_tipping.dm:7]:
```suggestion
		self_right_time = 15 SECONDS, \
```
As tempting as it is to lazily sit back on my bed and watch Skyrat alienate and burn off the last few borg players they've yet to get rid of, I still have a duty of care to the last two or three of them to quality control some of this stuff.
Letting some crybaby shitter give a borg a 60s stun if they stand still for a few seconds or start typing? No thanks, that can stay on TG.

File: modular_skyrat/modules/altborgs/code/robot_tipping.dm [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		post_tipped_callback = CALLBACK(src, .proc/after_tip_over), \
+		post_untipped_callback = CALLBACK(src, .proc/after_righted))
+
+/datum/component/tippable/skyrat/Initialize(
+	tip_time = 3 SECONDS,
+	untip_time = 1 SECONDS,
+	self_right_time = 60 SECONDS,
```
Comment (KathrinBailey) [inline modular_skyrat/modules/altborgs/code/robot_tipping.dm:14]:
```suggestion
	self_right_time = 15 SECONDS,
```
see above

**Review by KathrinBailey — CHANGES_REQUESTED** [review]
I know this isn't your PR so you aren't to blame but I'm getting tired of all the crybully shit against borgs as of late so the stun can be made 15s here or I'll make it 15s elsewhere.

**Comment by CandleJaxx** [comment]
i unironically see no downside to the tipping- that is placed there so that malf borgs can be non-lethally detained, and it's the same amount as handcuffs. that should stay entirely

**Comment by desertmote** [comment]
My personal preference is that this PR keeps balance changes out of it. As it stands it adds visual effects to roombas and a component for having a visual icon_state instead of a y180- flip. Changes to the numbers can be made after this is merged. 💚

**Review by KathrinBailey — APPROVED** [review]
will do the unflip changes myself separately, that way no one can @ you with salt-fuelled aggression

**Pull request merged**